### ISSUE TITLE
Allow a nullable column to be used as primary key/sorting key

### ISSUE THREAD

**Issue by silviucpp** [yandex/ClickHouse#5319]
Hello,

Currently clickhouse is not able to use as primary key/sorting key a nullable column.

If you try this you receive: "Sorting key cannot contain nullable columns".

I think in lot of use-cases the columns are nullable as time this is the standard for specifying that
a value doesn't exist. So lifting this limitation might be important.

I know there is another discussion here: https://github.com/yandex/ClickHouse/issues/4332 but this task was closed
for some reasons by the reporter.

Thanks,
Silviu

**Comment by alexey-milovidov** [comment]
Hi! This is doable but multiple potential complications exist.

Relevant parts of code:
`KeyCondition`, `FieldVisitorAccurateLess`, `FieldVisitorAccurateEquals`

**Comment by alexey-milovidov** [comment]
As an example, we allowed to use Nullables for minmax secondary index, but some issues start to appear #5246

**Comment by calebeaires** [comment]
Hey folks of Clickhouse! Great job!

Is this solution implemented? If so, who to enable it?

**Comment by yanchaoguo** [comment]
Hello, although version 20.7 solves this problem when creating a table, the new materialize MySQL feature automatically creates a table. If there is a null index in mysql, an error will still be reported

**Comment by filimonov** [comment]
@alexey-milovidov @amosbird  are all the parts done in #12433? Can we close that?

**Comment by alexey-milovidov** [comment]
No. Index analysis is not implemented for IS NULL, IS NOT NULL.

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/ClickHouse/pull/12455
